PalindromeChecker/OptimizedPalindromeChecker.java

```diff
@@ -0,0 +1,16 @@
+//Fastest runninig algorithm for Plindrome checker and optimized too
+
+class Solution {
+public boolean isPalindrome(int x) {
```

Comment:
This code works for numbers but the previous is for strings. Please do add a new file instead modifying as these 2 are different functionalities.